button-merchant/src/main/java/com/usebutton/merchant/PostOrderTask.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * PostOrderTask.java
+ *
+ * Copyright (c) 2019 Button, Inc. (https://usebutton.com)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+package com.usebutton.merchant;
+
+import android.support.annotation.Nullable;
+
+/**
+ * Asynchronous task used to report order to the Button API.
+ */
+class PostOrderTask extends Task {
+
+    private final ButtonApi buttonApi;
+    private final String applicationId;
+    private final String sourceToken;
+    private final Order order;
+    private final DeviceManager deviceManager;
+
+    PostOrderTask(@Nullable Listener listener, ButtonApi buttonApi, Order order,
```

Comment:
nit: I would move the nullable listener to end of the argument list

Reply:
Just being consistent at this point with the other tasks